src/config/configure.in

```diff
@@ -2619,7 +2673,7 @@ AS_IF([test x"$hypre_using_hip" == x"yes"],
         dnl interprets as C and therefore invokes the C compiler rather than the HIP part
         dnl of clang. Put HIPCXXFLAGS at the end so the user can override from
         dnl from the configure line.
-        HIPCXXFLAGS="-x hip -std=c++14 ${HIPCXXFLAGS}"
+        HIPCXXFLAGS="-x hip -fPIC -std=c++14 ${HIPCXXFLAGS}"
```

Comment:
Just curious why this is needed for AMD GPUs. I thought -fPIC was something to do with shared libraries.

Reply:
Thanks for checking. This was the solution proposed by an user. However, upon further investigation, it seems that using `-fPIE` is more appropriate since modern Linux distributions use position independent executable (PIE) by default during linking (even when dealing with static libraries).  Adding `-fPIE` to `hipcc` ensures that the device objects are compatible with the code relocation model used during linking phase. I updated the code to use `-fPIE`